packages/ra-core/src/form/choices/useChoicesContext.ts

```diff
@@ -4,14 +4,20 @@ import { useList } from '../../controller';
 import { ChoicesContext, ChoicesContextValue } from './ChoicesContext';
 
 export const useChoicesContext = <ChoicesType extends RaRecord = RaRecord>(
```

Comment:
I'm surprised this doesn't trigger a TS error as `RaRecord` can't be a `string`. I believe this is because `ChoicesProps` in `useChoices` accepts `any[]` for `choices`. This might be an issue later

Reply:
We can't type the choices as users may pass anything and set the `optionText` and `optionValue`. It's never been an issue ;)

Reply:
We should not restrict the `ChoicesType` then but let's do that in another PR